src/utils/Utils.js

```diff
@@ -415,3 +416,41 @@ export function sortToParams(sort) {
     .filter((k) => sort[k] !== undefined)
     .map((k) => `${sort[k]}${k}`);
 }
+
+/**
+ * Retrieves an array of objects from the provided array where a specific key matches a given value,
+ * or where the key does not exist.
+ * @param {Array<Object>} array The array of objects to search.
+ * @param {string} key The key to check within each object in the array.
+ * @param {any} value The value to compare against the value associated with the key in each object.
+ * @param {boolean} [keyMustExist=true] Flag indicating whether the specified key must exist in the object.
+ *                                      Defaults to true. If set to false, objects without the specified key will also be included.
+ * @returns {Array<Object>} An array containing all objects from the provided array
+ *                          where the specified key matches the provided value, or where the key does not exist.
+ */
+export function filterObjectsByKeyValuePair(array, key, value, keyMustExist = true) {
```

Comment:
Strange name/description of the function as it seems like general function, but at the same time it uses constant:
```
SConstants.HAS_SUBQUESTION
```

Reply:
Yes, I was even considering moving it somewhere else than utils as it is using constant... Do you have recommendation?

Reply:
Well,  I was not reading the function :( so I do not know :) (I have 2 mins to go somewhere :)

It is quite a complicated function at first sight compared to the description in the comment  ... If you change the comment to the function, then I will make my recommendation :)

Reply:
I refactored so it is clearer :)

Reply:
Ok I am not sure that i implemented what you wanted. But to me it seems that your solution would work only on 2 levels of questions, i.e.:
```
Section 1
  Question 1.1
```

But it would not work on 3 or more levels of questions:
```
Section 1
  Section 1.1
      Question 1.1.1
```

https://github.com/kbss-cvut/record-manager-ui/pull/129